builder/bundles.go

```diff
@@ -886,20 +892,59 @@ func fixOSRelease(filename, version string) error {
 		_ = f.Close()
 	}()
 
-	// TODO: If this is a mix, NAME and ID should probably change too. Create a section in
-	// configuration that will be used as reference to fill this.
-	// TODO: If this is a mix, add extra field for keeping track of the Clear Linux version
-	// used. Maybe also put the UPSTREAM URL, so we are ready to support mixes of mixes.
-	//
-	// See also: https://github.com/clearlinux/mixer-tools/issues/113
+	// Setup the osIDMap
+	osID := make(osIDMap)
+
+	var buf bytes.Buffer
+	var strMatch string
+
+	if err := toml.NewEncoder(&buf).Encode(config.Os); err != nil {
+		return err
+	}
+
+	// Prepare map table with TOML keys after encode which can be
+	// used for apple to apple comparison with default os-release file
+	re := regexp.MustCompile(`.*\".*\"`)
+	matches := re.FindAllStringSubmatch(buf.String(), -1)
+	for _, s := range matches {
+
+		//Remove " in string
+		strMatch = strings.Replace(s[0], "\"", "", -1)
+		//Split the string with "="
+		strSplit := strings.Split(strMatch, "=")
+		//Remove space in string
+		strSplit[0] = strings.Replace(strSplit[0], " ", "", -1)
+		strSplit[1] = strings.Replace(strSplit[1], " ", "", 1)
+		//Assign value into map table
+		osID[strSplit[0]] = strSplit[1]
+	}
+
+	// Add VERION_ID into map before os-release file modification is performed
+	osID["VERSION_ID"] = version
```

Comment:
We thought of creating custom os-release file also, just concern that need to create additional new file and thought of utilizing the builder.conf might be a good idea. 

Of course, there are some downsides for using builder.conf such as modification is need if we want to support new OS param and the string manipulation error risk that you mentioned. 
I think the string manipulation error risk can be low if we do code and validate properly.  For new OS param issue, I can add all the supported OS param by https://www.freedesktop.org/software/systemd/man/os-release.html. But that will create a long list of the [OsRelease] list. What do you think? 

I can go alternative if you are not agree with my suggestion.

Reply:
Just adding the values on that link doesn't solve that problem. os-release can contain any key-pair value provided that they respect shell formatting.

The idea is not to create a new file dedicated to set those values, but rather allow the user to provide the path to a file that will be replaced "as-is". For a normal mix, this path would be empty and no file would need to be provided.

Today it is already possible to change os-release file between "build bundles" and "build update". The problem is that unlike other custom files, mixer actually set the version value inside this file during "build bundles", which means that for this manual replacement to work the user must also manually set the version. The idea here is just to facilitate this process, letting mixer set the version on the custom file during "build bundles" step.

Reply:
Okay. I would like to align with you regarding the modification needed for alternative way. 

1. Enable builder.conf to support new OSRelease file as following format. Default is empty mean use the default os-release file. 
[OSRelease]
CUSTOM_OS_RELEASE = ""

2. Change the builder.conf format version from 1.0 -> 1.1. 

3. Replace the os-release file if custom path exist
- Perform file replacement inside fixOSRelease function if custom path provided inside builder.conf

Reply:
> 1. Enable builder.conf to support new OSRelease file as following format. Default is empty mean use the default os-release file.
>    [OSRelease]
>    CUSTOM_OS_RELEASE = ""

We don't need a new section for this. Add a new value to [Mixer] session and call it OS_RELEASE_PATH to keep the naming convention.

> 2. Change the builder.conf format version from 1.0 -> 1.1.

Correct. Conversion should work out of the box for this change, just make sure you add the new config to `bat/tests/config-conversion/configs/` as well. Name it `1.1_builder.conf`. The test automatically run on all configs on this folder.

> 3. Replace the os-release file if custom path exist
> 
> * Perform file replacement inside fixOSRelease function if custom path provided inside builder.conf

You can either do it inside fixOSRelease() or create a new function and call it inside buildOSCore(). Both would work just fine. You will need a reverence to the config in both cases. It is better to pass a reference to the builder object and access it with `b.Config` instead of using the config object directly. This will keep the changes in sync with the rest of the code and facilitate future refactoring.

There is also step 4: create a bat test to test the new feature.

Reply:
Okay. Thanks.